zkevm-circuits/src/ecc_circuit.rs

```diff
@@ -0,0 +1,1383 @@
+        // - P == (0, 0) and/or Q == (0, 0)
+        // - P == -Q, i.e. P + Q == R == (0, 0)
+        let res = op.r.unwrap_or(G1Affine::identity());
+        let point_r = self.handle_g1(ctx, ecc_chip, res, powers_of_256);
+        let rx_is_zero = ecc_chip.field_chip.is_zero(ctx, &point_r.ec_point.x);
+        let ry_is_zero = ecc_chip.field_chip.is_zero(ctx, &point_r.ec_point.y);
+
+        let rand_point = ecc_chip.load_random_point::<G1Affine>(ctx);
+        let point_p_is_zero = ecc_chip.field_chip.range().gate().or_and(
+            ctx,
+            QuantumCell::Existing(inputs_invalid),
+            QuantumCell::Existing(px_is_zero),
+            QuantumCell::Existing(py_is_zero),
+        );
+        let point_q_is_zero = ecc_chip.field_chip.range().gate().or_and(
+            ctx,
+            QuantumCell::Existing(inputs_invalid),
+            QuantumCell::Existing(qx_is_zero),
+            QuantumCell::Existing(qy_is_zero),
+        );
+        let point_r_is_zero = ecc_chip.field_chip.range().gate().or_and(
+            ctx,
+            QuantumCell::Existing(inputs_invalid),
+            QuantumCell::Existing(rx_is_zero),
+            QuantumCell::Existing(ry_is_zero),
+        );
+
+        // sum1 = if P == (0, 0) then r else r + P
+        let sum1 = ecc_chip.add_unequal(ctx, &rand_point, &point_p, true);
+        let sum1 = ecc_chip.select(ctx, &rand_point, &sum1, &point_p_is_zero);
+
+        // sum2 = if Q == (0, 0) then sum1 else sum1 + Q
+        let sum2 = ecc_chip.add_unequal(ctx, &sum1, &point_q, true);
+        let sum2 = ecc_chip.select(ctx, &sum1, &sum2, &point_q_is_zero);
+
+        // sum3 = if R == (0, 0) then sum2 else sum2 - R
+        let sum3 = ecc_chip.sub_unequal(ctx, &sum2, &point_r.ec_point, true);
+        let sum3 = ecc_chip.select(ctx, &sum2, &sum3, &point_r_is_zero);
+
+        ecc_chip.assert_equal(ctx, &rand_point, &sum3);
+
+        log::trace!("[ECC] EcAdd Assignmnet END:");
+        log_context_cursor!(ctx);
+
+        EcAddDecomposed {
+            is_valid: inputs_valid,
+            point_p: G1Decomposed {
+                ec_point: point_p,
+                x_cells: px_cells,
+                y_cells: py_cells,
+            },
+            point_q: G1Decomposed {
+                ec_point: point_q,
+                x_cells: qx_cells,
+                y_cells: qy_cells,
+            },
+            point_r,
+        }
+    }
+
+    /// Decomposes an EcMul operation to return each G1 element as cells representing its byte
+    /// form.
+    #[allow(clippy::too_many_arguments)]
+    fn decompose_ec_mul_op(
+        &self,
+        ctx: &mut Context<F>,
+        ecc_chip: &EccChip<F, FpConfig<F, Fq>>,
+        fr_chip: &FpConfig<F, Fr>,
+        _pairing_chip: &PairingChip<F>,
+        _fp12_chip: &Fp12Chip<F, FpConfig<F, Fq>, Fq12, XI_0>,
+        powers_of_256: &[QuantumCell<F>],
+        op: &EcMulOp,
+    ) -> EcMulDecomposed<F> {
+        log::trace!("[ECC] ==> EcMul Assignmnet START:");
```

Comment:
Nitpick: Assignment

Reply:
fixed in [4d5070f](https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/1720/commits/4d5070fdf1d7821cab0b4b2caf5f0582fe291ae2)